apps/next-evans/pages/friend/[friend_slug].tsx

```diff
@@ -1,36 +1,30 @@
 import { PrismaClient } from '@prisma/client';
+import invariant from 'tiny-invariant';
 import type { GetStaticPaths, GetStaticProps } from 'next';
-import type { Lang } from '@/../../libs-ts/types/src';
 import FeaturedQuoteBlock from '@/components/pages/friend/FeaturedQuoteBlock';
 import FriendBlock from '@/components/pages/friend/FriendBlock';
 import TestimonialsBlock from '@/components/pages/friend/TestimonialsBlock';
+import { LANG } from '@/lib/env';
 
 export const getStaticPaths: GetStaticPaths = async () => {
-  const prisma = new PrismaClient();
-  const allFriends = await prisma.friends.findMany();
+  const allFriends = await new PrismaClient().friends.findMany({
+    where: { lang: LANG },
+    select: { slug: true },
+  });
+
   const paths = allFriends.map((friend) => ({
-    params: {
-      friend_name: friend.slug,
-    },
+    params: { friend_slug: friend.slug },
   }));
 
-  return {
-    paths,
-    fallback: false,
-  };
+  return { paths, fallback: false };
 };
 
 export const getStaticProps: GetStaticProps<Props> = async (context) => {
-  const prisma = new PrismaClient();
-  if (typeof context.params?.friend_name !== `string`) {
-    return {
-      notFound: true,
-    };
-  }
-  const friend = await prisma.friends.findFirst({
+  invariant(typeof context.params?.friend_slug === `string`);
```

Comment:
i brought in a little library called `tiny-invariant`. it's perfect for this use case. we want to throw and error and fail the build in certain situations, but they should almost never happen, and we want the build to fail if they do. but we also need typescript to understand we're covering the error/undefined case. this handles both, and is much more succinct.

it works by using a type assertion `function invariant(condition: any) asserts condition`, so that's why TS is ok below the line.